### PR TITLE
rename precision from Laplace distribution to scale

### LINKED ISSUE #293: ‘precision’ paramter in Laplace should be called ‘rate’ such that is not confused with inverse of variance

### PR DESCRIPTION
**Pull request by chaozg**
Closes #293 by using `scale`, instead of `prec` to initialize Laplace `cuqi.distribution.Laplace`

### CODE REVIEW

**Review by nabriis — APPROVED** [review]
LGTM

**Review by amal-ghamdi — COMMENTED** [review]
Thank you @chaozg for your update! This looks good to me, I only have few comments:

- We can update also prec to rate in (demo)[https://github.com/CUQI-DTU/CUQIpy/blob/b856976b51fac225cc815ff2c54e3d32d45584da/demos/demo15_test_LMRF.py#L20] for consistency (it is not a bug)
- We might consider using scale (inverse of rate) instead of rate? to be consistent with LMRF (also it is the name numpy and spicy use)
- Would you please add closes #<issue number> to the PR description, this will allow linking those together :)

Many thanks!

**Comment by chaozg** [comment]
In the end, we moved from `prec` to `scale` in iniatilizing `cuqi.distribution.Laplace` to keep consistent with
- CUQIpy itself (`cuqi.distribution.LMRF`)
- Numpy's [Laplace](https://numpy.org/doc/stable/reference/random/generated/numpy.random.laplace.html)
- CUQI-PyTorch's [Laplace](https://github.com/CUQI-DTU/CUQIpy-PyTorch/blob/b10e4883f37a975da3d009fa9c47916ecbaabaac/cuqipy_pytorch/distribution.py#L399)

**Review by nabriis — APPROVED** [review]

**Review by amal-ghamdi — COMMENTED** [review]
Thank you @chaozg for your PR! This is really nice. I added one comment, and I have one additional minor comment here:

I just made a quick search of our repositories and I found:
- in this file demos/try12_Deconvolution_BayesianProblem.py we have commented out code that uses Laplace, it will be nice if you can fix it to pass scale instead of prec. (I think it is just should be `TP.prior = Laplace(np.zeros(n), 1/10)` instead of `TP.prior = Laplace(np.zeros(n), 10)`). And you can keep the code commented out.

File: demos/demo15_test_LMRF.py [hunk 1]
```diff
@@ -6,26 +6,27 @@
...
 print(y.pdf(6))
 print(y.pdf(1))
 samples = y.sample(1000)
 
-#%%
+# %%
 from scipy.stats import laplace
 print(laplace.pdf(6))
 print(laplace.pdf(1))
+assert np.isclose(y.pdf(6), laplace.pdf(6)) and np.isclose(y.pdf(1), laplace.pdf(1))
```
[the remark this answers is not in the document]
Comment (chaozg) [reply demos/demo15_test_LMRF.py]: Thank you @amal-ghamdi for reviewing the PR. As you suggested, I just put a new function `test_Laplace_pdf`  in `tests/test_distributions.py` for this purpose.

**Comment by chaozg** [comment]
Thank you @amal-ghamdi for pointing out `demos/try12`. I just did the update it as you advised and left that line commented out.

File: demos/demo15_test_LMRF.py [hunk 2]
```diff
@@ -6,23 +6,23 @@
...
+y = cuqi.distribution.Laplace(location, scale)
 print(y.pdf(6))
 print(y.pdf(1))
 samples = y.sample(1000)
 
-#%%
+# %%
 from scipy.stats import laplace
 print(laplace.pdf(6))
 print(laplace.pdf(1))
```
Comment (amal-ghamdi) [inline demos/demo15_test_LMRF.py:28]: Just a minor point here, these lines can be removed as well :)
```suggestion
```

File: tests/test_distribution.py [hunk 3]
```diff
@@ -372,6 +372,13 @@ def my_pdf( x, a, location, scale):
     else:
         raise ValueError
 
+@pytest.mark.parametrize("location", [-1, -2, -3, 0, 1, 2, 3])
+@pytest.mark.parametrize("scale", [1e-3, 1e-1, 1e0, 1e1, 1e3])
+@pytest.mark.parametrize("value", [1e-3, 1e-1, 1e0, 1e1, 1e3])
+def test_Laplace_pdf(location, scale, value):
+    LPL = cuqi.distribution.Laplace(location, scale)
+    assert np.isclose(LPL.pdf(value), scipy_stats.laplace(location, scale).pdf(value))
+
```
Comment (amal-ghamdi) [inline tests/test_distribution.py:381]: really nice tests 😊

**Review by amal-ghamdi — APPROVED** [review]
Hi @chaozg , Thank you so much for the update. I just made a minor suggestion, you can accept it directly if you like. I am approving the PR, it is good to go from my side.

You can re-request @nabriis review for now, because this is the workflow we agreed on last time. And we can discuss it in our next retro if we need to update the workflow 😊

**Pull request merged**